### PR TITLE
Feature/ftp transport protocol

### PR DESCRIPTION
**Pull request by gkiran292**

### CODE REVIEW

File: admin/.classpath [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (DImuthuUpe) [inline admin/.classpath:1]: Remove this file

File: agent/src/main/java/org/apache/airavata/mft/agent/MFTAgent.java [hunk 2]
```diff
@@ -100,94 +100,92 @@ public void init() {
```
Comment (DImuthuUpe) [inline agent/src/main/java/org/apache/airavata/mft/agent/MFTAgent.java:100]: Did you do auto formatting in this file? And what is the relevance of this change to ftp transport? If you are doing any change to agent or core, please raise them as separate PRs

File: agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java [hunk 3]
```diff
@@ -49,9 +49,9 @@ public void destroy() {
         executor.shutdown();
     }
 
-    public String transfer(TransferCommand command, Connector inConnector, Connector outConnector, MetadataCollector srcMetadataCollector,
```
Comment (DImuthuUpe) [inline agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java:52]: Remove this change. Not related to this PR

File: agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java [hunk 4]
```diff
@@ -82,140 +82,133 @@ public String transfer(TransferCommand command, Connector inConnector, Connector
 
 
         // Monitoring the completeness of the transfer
-        Thread monitorThread = new Thread(new Runnable() {
```
Comment (DImuthuUpe) [inline agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java:85]: Lots of unrelated changes. Do not do auto formatting from IDE and always look for the diff before committing.

File: agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java [hunk 5]
```diff
@@ -82,140 +82,133 @@ public String transfer(TransferCommand command, Connector inConnector, Connector
...
-                    }
-                } catch (Exception e) {
+                    double time = (endTime - startTime) * 1.0 / 1000000000;
 
                     statusLock.lock();
                     onStatusCallback.accept(command.getTransferId(), new TransferState()
-                            .setPercentage(0)
-                            .setState("FAILED")
+                            .setPercentage(100)
+                            .setState("COMPLETED")
```
Comment (DImuthuUpe) [inline agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java:158]: Why do you mark FAILED status to COMPLETED at exceptions?

File: agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java [hunk 6]
```diff
@@ -82,140 +82,133 @@ public String transfer(TransferCommand command, Connector inConnector, Connector
...
                             .setUpdateTimeMils(System.currentTimeMillis())
-                            .setDescription("Transfer failed due to " + ExceptionUtils.getStackTrace(e)));
+                            .setDescription("Transfer successfully completed"));
```
Comment (DImuthuUpe) [inline agent/src/main/java/org/apache/airavata/mft/agent/TransportMediator.java:160]: Wrong logic here

File: api/.project [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (DImuthuUpe) [inline api/.project:1]: Remove these hidden files. Not related to PR

File: api/client/.classpath [hunk 8]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (DImuthuUpe) [inline api/client/.classpath:1]: Junk files. Remove these

**Comment by gkiran292** [comment]
Since there are a lot of changes that I need to do creating a proper PR with only FTP changes. I will create the next PR after that is merged. I mistakenly did the formatting. Closing this PR as it is very difficult to edit here!
